src/sentry/grouping/__init__.py

```diff
@@ -1 +1,154 @@
+"""
+sentry.grouping
+~~~~~~~~~~~~~~~
+
+This package implements various grouping related functionality in Sentry.
+It's an evolution of an earlier grouping system and the backwards compatibility
+with that grouping system has created a certain amount of complexity.
+
+General Overview
+----------------
+
+Events are grouped together into issues by calculating hashes.  When a hash
+is already associated with a group that group is reused.  The grouping code
+can however generate more than one hash the sum of which will be added to a
+group.  In the database this is represented in the `GroupHash` model.
```

Comment:
This might require a little more explanation (potentially at the end of this section): The hash of which variant is chosen for grouping in the end? Given that each variant produces it's hash, each could associate the event to a different group?